airbyte-integrations/connectors/source-jira/source_jira/utils.py

```diff
@@ -17,16 +22,40 @@ def safe_max(arg1, arg2):
 
 
 def read_full_refresh(stream_instance: Stream):
-    slices = stream_instance.stream_slices(sync_mode=SyncMode.full_refresh)
-    for _slice in slices:
-        records = stream_instance.read_records(stream_slice=_slice, sync_mode=SyncMode.full_refresh)
-        for record in records:
-            yield record
+    yield from _read(stream_instance, SyncMode.full_refresh, _NO_STATE)
 
 
 def read_incremental(stream_instance: Stream, stream_state: MutableMapping[str, Any]):
-    slices = stream_instance.stream_slices(sync_mode=SyncMode.incremental, stream_state=stream_state)
-    for _slice in slices:
-        records = stream_instance.read_records(sync_mode=SyncMode.incremental, stream_slice=_slice, stream_state=stream_state)
-        for record in records:
+    yield from _read(stream_instance, SyncMode.incremental, stream_state)
+
+
+def _read(stream_instance: Stream, sync_mode: SyncMode, stream_state: MutableMapping[str, Any]):
```

Comment:
What is this code doing differently than `entrypoint_wrapper.read`? It feels like it would be less code to maintain and therefore better to use this

Reply:
I thought this was for testing purposes. Ignore this one. I think what we want is `Stream.read_only_records`. Would that make more sense?

Reply:
updated with `read_only_records()`, removed custom `_read()`